netebpfext/net_ebpf_ext_sock_addr.c

```diff
@@ -771,30 +969,36 @@ _net_ebpf_extension_sock_addr_copy_wfp_connection_fields(
     // Copy IP address fields.
     if ((hook_id == EBPF_HOOK_ALE_AUTH_CONNECT_V4) || (hook_id == EBPF_HOOK_ALE_AUTH_RECV_ACCEPT_V4) ||
         (hook_id == EBPF_HOOK_ALE_CONNECT_REDIRECT_V4)) {
-        sock_addr_ctx->family = AF_INET;
-        sock_addr_ctx->msg_src_ip4 = htonl(incoming_values[source_ip_address_field].value.uint32);
-        sock_addr_ctx->user_ip4 = htonl(incoming_values[destination_ip_address_field].value.uint32);
+        sock_addr_ctx->base.family = AF_INET;
+        sock_addr_ctx->base.msg_src_ip4 = htonl(incoming_values[source_ip_address_field].value.uint32);
+        sock_addr_ctx->base.user_ip4 = htonl(incoming_values[destination_ip_address_field].value.uint32);
     } else {
-        sock_addr_ctx->family = AF_INET6;
+        sock_addr_ctx->base.family = AF_INET6;
         RtlCopyMemory(
-            sock_addr_ctx->msg_src_ip6,
+            sock_addr_ctx->base.msg_src_ip6,
             incoming_values[source_ip_address_field].value.byteArray16,
             sizeof(FWP_BYTE_ARRAY16));
         RtlCopyMemory(
-            sock_addr_ctx->user_ip6,
+            sock_addr_ctx->base.user_ip6,
             incoming_values[destination_ip_address_field].value.byteArray16,
             sizeof(FWP_BYTE_ARRAY16));
     }
-    sock_addr_ctx->msg_src_port = htons(incoming_values[source_port_field].value.uint16);
-    sock_addr_ctx->user_port = htons(incoming_values[destination_port_field].value.uint16);
-    sock_addr_ctx->protocol = incoming_values[fields->protocol_field].value.uint8;
-    sock_addr_ctx->compartment_id = incoming_values[fields->compartment_id_field].value.uint32;
+    sock_addr_ctx->base.msg_src_port = htons(incoming_values[source_port_field].value.uint16);
+    sock_addr_ctx->base.user_port = htons(incoming_values[destination_port_field].value.uint16);
+    sock_addr_ctx->base.protocol = incoming_values[fields->protocol_field].value.uint8;
+    sock_addr_ctx->base.compartment_id = incoming_values[fields->compartment_id_field].value.uint32;
 
     if (hook_id == EBPF_HOOK_ALE_CONNECT_REDIRECT_V4 || hook_id == EBPF_HOOK_ALE_CONNECT_REDIRECT_V6) {
-        sock_addr_ctx->interface_luid = 0;
+        sock_addr_ctx->base.interface_luid = 0;
     } else {
-        sock_addr_ctx->interface_luid = *incoming_values[fields->interface_luid_field].value.uint64;
+        sock_addr_ctx->base.interface_luid = *incoming_values[fields->interface_luid_field].value.uint64;
     }
+
+    sock_addr_ctx->access_information =
+        (TOKEN_ACCESS_INFORMATION*)(incoming_values[FWPS_FIELD_ALE_CONNECT_REDIRECT_V4_ALE_USER_ID]
+                                        .value.byteBlob->data);
+    ASSERT(incoming_metadata_values->currentMetadataValues & FWPS_METADATA_FIELD_PROCESS_ID);
```

Comment:
If would check for the `FWPS_METADATA_FIELD` and initialize `process_id` to zero if meta info is missing.

Reply:
Fixed.